src/ripple/protocol/impl/TxFormats.cpp

```diff
@@ -322,6 +325,71 @@ TxFormats::TxFormats()
             {sfTicketSequence, soeOPTIONAL},
         },
         commonFields);
+
+    add(jss::XChainCreateBridge,
+        ttXCHAIN_CREATE_BRIDGE,
+        {
+            {sfXChainBridge, soeREQUIRED},
+            {sfSignatureReward, soeREQUIRED},
+            {sfMinAccountCreateAmount, soeOPTIONAL},
+        },
+        commonFields);
+
+    add(jss::XChainModifyBridge,
+        ttXCHAIN_MODIFY_BRIDGE,
+        {
+            {sfXChainBridge, soeREQUIRED},
+            {sfSignatureReward, soeOPTIONAL},
+            {sfMinAccountCreateAmount, soeOPTIONAL},
+        },
+        commonFields);
+
+    add(jss::XChainCreateClaimID,
+        ttXCHAIN_CREATE_CLAIM_ID,
+        {
+            {sfXChainBridge, soeREQUIRED},
+            {sfSignatureReward, soeREQUIRED},
+            {sfOtherChainSource, soeREQUIRED},
+        },
+        commonFields);
+
+    add(jss::XChainCommit,
+        ttXCHAIN_COMMIT,
+        {
+            {sfXChainBridge, soeREQUIRED},
+            {sfXChainClaimID, soeREQUIRED},
+            {sfAmount, soeREQUIRED},
+            {sfOtherChainDestination, soeOPTIONAL},
```

Comment:
I''m thinking the `sfOtherChainDestination` destination should be required.  Right now if that field is omitted (and the missing field is percolated through the attestation) then the resulting claim must be recovered with an `XChainClaim`.  The `XChainClaim` is used to recover from errors.  So making the `sfOtherChainDestination` field optional feels a bit like making it easy for users to create errors?

I'm not seeing a good user scenario for the `sfOtherChainDestination` field being optional.  Did I miss something?

Reply:
I don't see how this increases the probability that the user will use the transaction incorrectly. Deciding on a destination in the future seems like a reasonable thing to want, but it's not strictly needed. It's easy enough to remove this if people want tho. I'm weakly in favor of keeping it tho.

Reply:
> I don't see how this increases the probability that the user will use the transaction incorrectly.

My current view is that typical XChain transfers are from one account to a known account on the other chain.  As far as I can see it's not typical for someone to "just send money to the other chain and I'll figure out who gets the money once it's there."  For someone who wants to follow the typical path, where they know the money's destination, if they omit the `Destination` they loused up the transaction and they have to untangle it manually on the other chain.  That's what I mean by, "making it easy for users to create errors."

So what I'd like to see is a bona fide use case where omitting the `OtherChainDestination` serves some useful purpose.  I'm hoping for something a lot more specific than a vague, "well someone might find it useful".

In summary, please show me a good use case where omitting the `OtherChainDestination ` field in the transaction does something useful and desirable.  I can't see a use case like that from where I'm standing.

Reply:
Fixed in d1b93d78fe [fold] Require dst in commit txn: https://github.com/XRPLF/rippled/pull/4292#discussion_r1123839821

Reply:
I'm re-opening this. I did make `dst` required, but I think we really do want to allow that to be optional. If you want to ensure the amount was delivered no the dst chain, the user will want to submit that transaction themselves. I was "weakly against" removing this before, but I think I'm now "agaist" removing it. I'm going to add the code back in, but of course we can discuss this further.